packages/cw-pre-propose-base/src/state.rs

```diff
@@ -0,0 +1,71 @@
+use std::marker::PhantomData;
+
+use cosmwasm_std::Addr;
+use cw_storage_plus::{Item, Map};
+use serde::{Deserialize, Serialize};
+
+use voting::deposit::CheckedDepositInfo;
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
+pub struct Config {
+    /// Information about the deposit required to create a
+    /// proposal. If `None`, no deposit is required.
+    pub deposit_info: Option<CheckedDepositInfo>,
+    /// If false, only members (addresses with voting power) may create
+    /// proposals in the DAO. Otherwise, any address may create a
+    /// proposal so long as they pay the deposit.
+    pub open_proposal_submission: bool,
+}
+
+pub struct PreProposeContract<InstantiateExt, ExecuteExt, QueryExt, ProposalMessage> {
+    /// The proposal module that this module is associated with.
+    pub proposal_module: Item<'static, Addr>,
+    /// The DAO (cw-dao-core module) that this module is associated
+    /// with.
+    pub dao: Item<'static, Addr>,
+    /// The configuration for this module.
+    pub config: Item<'static, Config>,
+    /// Map between proposal IDs and (deposit, proposer) pairs.
+    pub deposits: Map<'static, u64, (Option<CheckedDepositInfo>, Addr)>,
```

Comment:
how will it have to change when allowing multiple people to contribute different amounts to deposits?

Reply:
we'll need a different data structure for that. :) can likely borrow what ion dao [has done](https://github.com/many-things/ion-dao-contracts/blob/dev/contracts/dao/src/state.rs#L80-L82).